cis194/week1/wei-ning/src/Exercise4.hs

```diff
@@ -0,0 +1,14 @@
+module Exercise4 (runTests) where
+
+import           Assert
+import           Exercise1 (toDigitsRev)
+import           Exercise2 (doubleEveryOther)
+import           Exercise3 (sumDigits)
+
+runTests :: IO ()
+runTests = do
+  assert "ex1.4 expect valid credit card num" $ validate 4012888888881881
+  assert "ex1.4 expect invalid credit card num" $ not (validate 4012888888881882)
+
+validate :: Integer -> Bool
+validate = (== 0) . (`mod` 10) . sumDigits . doubleEveryOther . toDigitsRev
```

Comment:
seems `mod` 10 is not correct. as well as toDigitsRev. is Rev necessary?
here is the description of the task:
```
Calculate the remainder when the sum is divided by 10. For the
above example, the remainder would be 8.
```

Reply:
oh, sorry actually, this describes `mod` 10